src/replaceCustomValues.ts

```diff
@@ -175,68 +175,155 @@ export function getElementSubmissionValue({
   formatNumber,
   formatCurrency,
 }: {
-  elementId?: string
-  propertyName?: string
+  propertyName: string
   formElements: FormTypes.FormElement[]
   submission: SubmissionTypes.S3SubmissionData['submission']
 } & ReplaceInjectablesFormatters):
   | {
       element: FormTypes.FormElement | undefined
       value: unknown
     }
-  | undefined {
+  | undefined
+export function getElementSubmissionValue({
+  elementId,
+  submission,
+  formElements,
+  formatDate,
+  formatDateTime,
+  formatTime,
+  formatNumber,
+  formatCurrency,
+}: {
+  elementId: string
+  formElements: FormTypes.FormElement[]
+  submission: SubmissionTypes.S3SubmissionData['submission']
+} & ReplaceInjectablesFormatters):
+  | {
+      element: FormTypes.FormElement | undefined
+      value: unknown
+    }
+  | undefined
+export function getElementSubmissionValue({
+  elementId,
+  propertyName,
+  submission,
+  formElements,
+  formatDate,
+  formatDateTime,
+  formatTime,
+  formatNumber,
+  formatCurrency,
+}: {
+  elementId?: string
+  propertyName?: string
+  formElements: FormTypes.FormElement[]
+  submission: SubmissionTypes.S3SubmissionData['submission']
+} & ReplaceInjectablesFormatters) {
   if (elementId === undefined && propertyName === undefined) {
     return undefined
   }
 
+  let result: ReturnType<typeof getElementSubmissionValueByName> | undefined =
+    undefined
+
+  if (elementId) {
+    result = getElementSubmissionValueById({
+      elementId,
+      formElements,
+      submission,
+    })
+  } else if (propertyName) {
+    result = getElementSubmissionValueByName({
+      propertyName,
+      formElements,
+      submission,
+    })
+  }
+
+  if (!result) {
+    return
+  }
+  return formatValue({
+    element: result.formElement,
+    unknownValue: result.unknownValue,
+    formatDate,
+    formatTime,
+    formatDateTime,
+    formatCurrency,
+    formatNumber,
+  })
+}
+
+function getElementSubmissionValueByName({
+  propertyName,
+  formElements,
+  submission,
+}: {
+  propertyName: string
+  formElements: FormTypes.FormElement[]
+  submission: SubmissionTypes.S3SubmissionData['submission']
+}):
+  | { formElement: FormTypes.FormElement | undefined; unknownValue: unknown }
+  | undefined {
+  const formElement = findFormElement(
+    formElements,
+    (element) =>
+      element.type !== 'page' &&
+      element.type !== 'section' &&
+      element.name === propertyName,
+  )
+
+  const unknownValue = submission[propertyName]
+  if (unknownValue === undefined || unknownValue === null) {
+    return undefined
+  }
```

Comment:
```suggestion
  const unknownValue = submission[propertyName]
  if (unknownValue === undefined || unknownValue === null) {
    return undefined
  }
  
  const formElement = findFormElement(
    formElements,
    (element) =>
      element.type !== 'page' &&
      element.type !== 'section' &&
      element.name === propertyName,
  )
```

Can avoid iterating over elements if we dont need to.